staging/src/k8s.io/kubectl/pkg/kuberc/kuberc.go

```diff
@@ -0,0 +1,408 @@
+	}
+
+	kubercPath := getExplicitKuberc(args)
+	kuberc, err := p.getPreferencesFunc(kubercPath)
+	if err != nil {
+		return args, fmt.Errorf("kuberc error %w", err)
+	}
+
+	if kuberc == nil {
+		return args, nil
+	}
+
+	err = validate(kuberc)
+	if err != nil {
+		return args, err
+	}
+
+	args, err = p.applyAliases(rootCmd, kuberc, args, errOut)
+	if err != nil {
+		return args, err
+	}
+	err = p.applyOverrides(rootCmd, kuberc, args, errOut)
+	if err != nil {
+		return args, err
+	}
+	return args, nil
+}
+
+// applyOverrides finds the command and sets the defaulted flag values in kuberc.
+func (p *Preferences) applyOverrides(rootCmd *cobra.Command, kuberc *config.Preference, args []string, errOut io.Writer) error {
+	args = args[1:]
+	cmd, _, err := rootCmd.Find(args)
+	if err != nil {
+		fmt.Fprintf(errOut, "Warning: command not found %v\n", err)
+		return nil
+	}
+
+	for _, c := range kuberc.Overrides {
+		parsedCmds := strings.Fields(c.Command)
+		overrideCmd, _, err := rootCmd.Find(parsedCmds)
+		if err != nil {
+			fmt.Fprintf(errOut, "Warning: command %q not found to set kuberc override\n", c.Command)
+			continue
+		}
+		if overrideCmd.Name() != cmd.Name() {
+			continue
+		}
+
+		if _, ok := p.aliases[cmd.Name()]; ok {
+			return fmt.Errorf("alias %s can not be overridden", cmd.Name())
+		}
+
+		// This function triggers merging the persistent flags in the parent commands.
+		_ = cmd.InheritedFlags()
+
+		for _, fl := range c.Flags {
+			existingFlag := cmd.Flag(fl.Name)
+			if existingFlag == nil {
+				return fmt.Errorf("invalid flag %s for command %s", fl.Name, c.Command)
+			}
+			if searchInArgs(existingFlag.Name, existingFlag.Shorthand, cmd.Flags(), args) {
+				// Don't modify the value implicitly, if it is passed in args explicitly
+				continue
+			}
+			err = cmd.Flags().Set(fl.Name, fl.Default)
+			if err != nil {
+				return fmt.Errorf("could not apply override value %s to flag %s in command %s err: %w", fl.Default, fl.Name, c.Command, err)
+			}
+		}
+	}
+
+	return nil
+}
+
+// applyAliases firstly appends all defined aliases in kuberc file to the root command.
+// Since there may be several alias definitions belonging to the same command, it extracts the
+// alias that is currently executed from args. After that it sets the flag definitions in alias as default values
+// of the command. Lastly, others parameters (e.g. resources, etc.) that are passed as arguments in kuberc
+// is appended into the command args.
+func (p *Preferences) applyAliases(rootCmd *cobra.Command, kuberc *config.Preference, args []string, errOut io.Writer) ([]string, error) {
+	_, _, err := rootCmd.Find(args[1:])
+	if err == nil {
+		// Command is found, no need to continue for aliasing
+		return args, nil
+	}
+
+	var aliasArgs *aliasing
+
+	var commandName string // first "non-flag" arguments
+	for _, arg := range args[1:] {
+		if !strings.HasPrefix(arg, "-") {
+			commandName = arg
+			break
+		}
+	}
+	commandName = strings.TrimSpace(commandName)
```

Comment:
TrimSpace seems wrong... when would the command name have leading / trailing spaces that we should ignore?

Reply:
I checked it and verified that Cobra already sanitizes this for us. Removed TrimSpace.